### PR TITLE
Fix dashboard and question actions visibility depending on user's permissions

### LINKED ISSUE #13229: Revision History shows Revert button when user only has view-only permissions
**Describe the bug**
Revert button is shown even when the user doesn't have permissions to modify question.

**To Reproduce**
1. Create a question, save as "Q1" to "Our Analytics", make a modification and save again
2. Admin > Permissions > Collections > set "Our Analytics" to view-only
3. Admin > People > create new user "U1"
4. Login as "U1" > Our Analytics > Q1 > click pencil-icon next to question title > View revision history > clicking Revert button will make the button show `Revert failed`:
image

**Expected behavior**
Revert button shouldn't be shown, when user doesn't have permissions to modify question.

**Information about your Metabase Installation:**
Metabase 0.36.4

### PR DESCRIPTION
**Pull request by kulyk**
Fixes #13229, fixes the missing parts of #15280 

Fixes dashboard and question actions visibility depending on user's permission.
The next items are now only visible to users with `write` permission:

* "Change title and description" for dashboards
* "Archive" action for dashboards
* "Revert" buttons for dashboards and questions revision histories. Users with `read` can see the whole revision history, but can't try to revery any revision

### Read-only revision history demo

image

### CODE REVIEW

**Review by nemanjaglumac — APPROVED** [review]
Ran tests locally - LGTM.
I'd still wait on Dalton for the frontend changes review before the merge, though.

**Review by daltojohnso — APPROVED** [review]
nice work

**Pull request merged**
